remove v1 v2 references

Crissman opened the pull request:
Remove deprecation notices and style change notices for v1 and v2 of Chainer as per #5026

kmaehashi commented on docs/source/reference/debug.rst:
```diff
@@ -34,13 +34,5 @@ You can check if debug mode is enabled with :func:`chainer.is_debug` function.
 Deprecated interface
```
I think the section itself can be removed.

kmaehashi replied:
This section is a list of "Deprecated interface", and there's nothing to list as `DebugMode` has been removed.
So I think it's OK to remove the section.

kmaehashi commented on chainer/link.py:
```diff
@@ -551,15 +551,6 @@ def cleargrads(self):
             param.cleargrad()
 
     def zerograds(self):
-        """Initializes all gradient arrays by zero.
```
I think it's better not to remove the existing docs for deprecated methods.
It will show up as an undocumented method in the docs, which may confuse users.

kmaehashi commented:
There are some others that need to be removed:

https://docs.chainer.org/en/latest/reference/generated/chainer.Deserializer.html?highlight=v2.0.0
https://docs.chainer.org/en/latest/reference/generated/chainer.serializers.HDF5Serializer.html?highlight=v2.0.0

Crissman commented:
Thanks, @kmaehashi for pointing out the missing old version notes. I had renamed my local branch, and so the commits were not updating this branch. Fixed.

As for removing existing docs for deprecated methods, I think it would be best to remove those methods entirely from the docs. I looked for a way to do this with Read the Docs, but couldn't find one. Is there a way to do this?

If not, I'll add comments to those functions that they're deprecated.

Thanks!

kmaehashi commented:
As for deprecated module members like `DebugMode` and `assert_allclose`, just remove the reference from docs (https://github.com/chainer/chainer/pull/5081/files#diff-d7aa349852b1ed75022a62937030d452L46). Let's leave the pydocs (docs in the source) so that users can still understand why they're undocumented by looking into the code.

As for deprecated class members like `Link.zerograds`, I think there's no way to hide it, unfortunately.

niboshi commented:
>As for deprecated class members like `Link.zerograds`, I think there's no way to hide it, unfortunately.

It's a bit tricky but maybe we can blacklist in `docs/source/_templates/autosummary/class.rst`?

Crissman commented:
@niboshi I checked, and blacklisting in `class.rst` works! But, I don't think we should do that given @kmaehashi 's comments about leaving the docs for `zerograds`, so I've left it for now.

I think this is ok to merge. Please confirm!

kmaehashi commented on chainer/backends/cuda.py:
```diff
@@ -196,7 +196,7 @@ def get_device(*args):
     .. note::
 
-        This API is deprecated. Please use
+        This API is deprecated since v3.0.0. Please use
```
It seems its already deprecated in v2.

Crissman replied:
Fixed in master already.

kmaehashi commented on chainer/link.py:
```diff
@@ -554,13 +554,11 @@ def cleargrads(self):
 
-        .. deprecated:: v1.15
-           Use :meth:`cleargrads` instead.
+         """
```
* Indentation seems broken.
* How about saying `Use :meth:`cleargrads` instead, which is more efficient.`? So users can understand this is not just a rename.

Crissman replied:
Updated.

kmaehashi commented on chainer/links/connection/mlp_convolution_2d.py:
```diff
@@ -54,10 +54,8 @@ class MLPConvolution2D(link.ChainList):
-        keyword arguments only. We impose this restriction to forbid
-        users to assume the API for v1 and specify `wscale` option,
-        that had been between `activation` and `conv_init` arguments in v1.
+        `conv_init` and `bias_init` arguments must be specified as
```
I think the entire note block can be removed, as there are many other keyword-arguments-only methods.

kmaehashi commented on chainer/links/connection/parameter.py:
```diff
@@ -7,10 +7,6 @@ class Parameter(link.Link):
 
     """Link that just holds a parameter and returns it.
 
-    .. deprecated:: v1.5
```
This is a documented API and thus cannot be removed.
https://docs.chainer.org/en/stable/reference/generated/chainer.links.Parameter.html?highlight=Parameter

kmaehashi commented on chainer/optimizer.py:
```diff
@@ -690,21 +690,6 @@ def update(self, lossfun=None, *args, **kwds):
         self.call_hooks('post')
 
     def use_cleargrads(self, use=True):
-        """Enables or disables use of :func:`~chainer.Link.cleargrads` in `update`.
```
This is a documented api and thus cannot be removed.
https://docs.chainer.org/en/stable/reference/generated/chainer.GradientMethod.html#chainer.GradientMethod

kmaehashi commented on chainer/variable.py:
```diff
@@ -805,16 +800,6 @@ def cleargrad(self):
         self._grad_var = None
 
     def zerograd(self):
-        """Initializes the gradient array by zeros.
```
This is an documented API. https://docs.chainer.org/en/stable/reference/generated/chainer.Variable.html#chainer.Variable.zerograd

How about fixing this as you did in Link?
https://github.com/chainer/chainer/pull/5081/files#diff-349b58823620f69f6f4ad574377a654d

Crissman commented:
@kmaehashi Please confirm review ok, thanks.

kmaehashi commented on chainer/__init__.py:
```diff
@@ -229,21 +229,6 @@ def set_debug(debug):
 
 
 class DebugMode(object):
-    """Debug mode context.
```
https://github.com/chainer/chainer/pull/5081#issuecomment-406478374

kmaehashi commented on chainer/backends/cuda.py:
```diff
@@ -266,10 +266,9 @@ def get_device_from_array(*arrays):
 
-        This API is deprecated. Please use
-        :func:`~chainer.backends.cuda.get_device_from_id`
+        Please use :func:`~chainer.backends.cuda.get_device_from_id`
```
This docs should not be changed as it is listed on the reference.

kmaehashi commented on chainer/gradient_check.py:
```diff
@@ -290,18 +290,6 @@ def iterate_single_input(i_in, x, orig_x, i):
 
 
 def assert_allclose(x, y, atol=1e-5, rtol=1e-4, verbose=True):
-    """Asserts if some corresponding element of x and y differs too much.
```
https://github.com/chainer/chainer/pull/5081#issuecomment-406478374

kmaehashi commented on chainer/optimizer.py:
```diff
@@ -757,16 +757,16 @@ def use_cleargrads(self, use=True):
-           :meth:`~Link.zerograds`, so one does not have to call
-           :meth:`use_cleargrads`. This method remains for backward
-           compatibility.
+             use (bool): If ``True``, this function enables use of
```
Space unexpectedly inserted?

Crissman commented:
Review addressed. Please confirm, @kmaehashi -san!

kmaehashi commented on chainer/optimizer.py:
```diff
@@ -761,12 +761,12 @@ def use_cleargrads(self, use=True):
-           :meth:`~Link.zerograds`, so one does not have to call
-           :meth:`use_cleargrads`. This method remains for backward
-           compatibility.
+         .. deprecated:: v2.0
```
Space unexpectedly inserted?

Crissman commented:
Fixed! Thanks. @kmaehashi

kmaehashi commented:
LGTM! Jenkins, test this please.

kmaehashi commented:
LGTM! Test failures are not related to this PR.
